internal/meta/db.go

```diff
@@ -0,0 +1,26 @@
+package meta
+
+type DB interface {
+	ReadTx
+	// WithTx runs the given function in a transaction. The transaction is
+	// committed when the function returns (unless it has been aborted by
+	// calling WriteTx.Abort).
+	WithTx(func(tx WriteTx)) error
```

Comment:
`WithTx(func(tx WriteTx) error)) error` might be more convenient as we can use it in RunE directly.

```
RunE: func(...) error {
    db := ...
    return db.WriteTx(func(tx WriteTx) error) error {
        ...
    })
}
```

Reply:
That was actually my original strategy but I chose against it to make it more clear that it's not a direct passthrough -- `WriteTx` can returns its own errors. Honestly, in hindsight, I think just doing a
```go
tx := db.WriteTx()
tx.SetBranch(...)
tx.Commit()
```
will be a better API. I chose the function-arg version to make it impossible to forget to commit/abort a transaction, but I think it's fine to just have an API invariant of "you MUST call `Commit` or `Abort` on this transaction object".